db/migrate/20240527070439_add_project_query_roles.rb

```diff
@@ -0,0 +1,21 @@
+class AddProjectQueryRoles < ActiveRecord::Migration[7.1]
+  def up
+    view_role ||= ProjectQueryRole.find_or_initialize_by(builtin: Role::BUILTIN_PROJECT_QUERY_VIEW)
+
+    view_role.update!(
+      name: I18n.t("seeds.common.project_query_roles.item_0.name", default: "Project query viewer"),
+      permissions: %i[
+        view_project_query
+      ]
+    )
+
+    edit_role ||= ProjectQueryRole.find_or_initialize_by(builtin: Role::BUILTIN_PROJECT_QUERY_EDIT)
+    edit_role.update!(
+      name: I18n.t("seeds.common.project_query_roles.item_1.name", default: "Project query editor"),
+      permissions: %i[
+        view_project_query
+        edit_project_query
+      ]
+    )
+  end
```

Comment:
This is an unsolved question on my end. Doing this will have lead to the roles being written to the database. And we definitely want to have that for installations that just do a migration from say 14.0 to the current one. But it will do the same for new instances. And that renders the seeding of the roles worthless.

Reply:
We had the same discussion when we introduced the roles for WorkPackages. First we planned to do it with seeds only, but later we still added them via a migration as well.

As we have a unique identfier with the builtin column being filled, the migration and the seeds would be idempotent as they would lead to the same result. 

I have no problem removing them from the seeds, I just think it's more complete to have them in both.